dotm_search.py

```diff
@@ -1,15 +1,50 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+import os
+import argparse
+import zipfile
+parser = argparse.ArgumentParser()
+
 
 """
 Given a directory path, search all files in the path for a given text string
 within the 'word/document.xml' section of a MSWord .dotm file.
 """
-__author__ = "???"
+__author__ = "knmarvel"
+#   time spent on this project: 2 hrs
+
+
+def examine_files(dir, char):
+    files = {}
+    total_files_searched = 0
+    for filename in os.listdir(dir):
+        filepath = dir + "/" + filename
+        if zipfile.is_zipfile(filepath):
+            with zipfile.ZipFile(filepath, 'r') as zip_ref:
+                zip_ref.extractall("./extracted")
+            with open('./extracted/word/document.xml', "r") as doc:
+                doc = doc.read()
+            if "office/word" in doc:
+                total_files_searched += 1
+                if char in doc:
+                    files[filename] = doc[doc.index(char) - 40: doc.index(char) + 40]
+    f_srched = "# of files searched: " + str(total_files_searched)
+    f_fnd = "# of files found:" + str(len(files))
+    f = "\n".join([": ".join([x, files[x]]) for x in files])
```

Comment:
i tried to make sense of this line 5 times... and couldn't.  i'm guessing that this is the list of the files and the context surrounding the search term?  is there a way that this could be written more clearly?

Reply:
I tried to split this up, but it didn't make it more clear. 
I join key value pairs in a dictionary into a string so they look like "Filename: String Context"
I turn those key value pairs into a list with list comprehension.
I join those list objects into a string with a new line.

Reply:
how about something along the lines of...

results = []
for file in files:
    results.append(str(file.filename) + ": " + sr(file.context))

i know this isn't exactly your variable names... but that kind of idea?

Reply:
list concat is totally "pythony"... i'm just trying to think how to take some complexity out of the line.

Reply:
I made the variable "x" in to the variable "filename" which I hope makes more sense.